src/main.ts

```diff
@@ -120,8 +122,13 @@ async function main() {
 	});
 
 	get('/block/:number', async (params) => {
-		const number = parseNumber(params.number);
-		const hash = await api.rpc.chain.getBlockHash(number);
+		let hash: BlockHash;
+		if (isHex(params.number)) {
```

Comment:
So, I know with txwrapper, you can construct an unsigned transaction with the block number in hex, like '0x1234'. Makes me wonder two things:
1. Some users might just have hex everything, in which case this should also check that the hex is 32 bytes.
2. Maybe Polkadot JS already has a nice function for this logic?

Reply:
Yeah, totally good point :+1:

Reply:
Yeah correct. I'm not aware of any helper function in polkadot-js, but 32-byte check should be fine.

There's an edge case where the hex-encoded number is 32-bytes, but this should happen in the very distant future, so no worries. Another approach, without disambiguation, would be to have another endpoint `/blockByHash/:hash`